### DISCUSSION TITLE
Feature proposal - Generate release notes automatically via the CD pipeline when deploying builds to Firebase/TestFlight

### DISCUSSION THREAD

**Discussion by minhnimble** [nimblehq/ios-templates#481] [Polls]
## Issue

During the working time on the client project, there has been a request from the PM to have release notes available on each created build to let users know the new changes included in that build rather than providing only the build number. Unfortunately, the iOS template doesn't support it on our CD pipeline to automatically generate a release note, and we had to work around it by leaving the release notes empty or adding the note manually case-by-case.

While this matter can sound trivial and look like a nice-to-have feature, this can be helpful for external testers and even clients who can expect and gain insights into the features implemented in the given build without having to double-check with our team every time testing the app. Hence, this can be a great time to consider implementing it. 

## Solution

The goal is to support generating release notes for both of the following cases - when distributing a new build after merging a PR to `develop` (after each story is completed) and after merging a release PR to `main` (at the end of the sprint) by reusing the merge commit messages:
- [after merging a PR to](https://github.com/nimblehq/tpc-delivery-android/blob/1c53bb393d0ce842bd6b46e292d723b20793c0fc/codemagic.yaml#LL174C17-L174C37) develop:
  - Command: `echo "RELEASE_NOTE_CONTENT="$((git log -1 --merges | grep "\[") | grep . && echo "" || echo $(git log -1 --merges --format=%B))""`
  - Sample release notes: 
    ```
    Merge pull request #172 from nimblehq/feature/sc-64238-ui-as-a-user-i-can-see-the-updated-color
    ```
- [after merging a release PR to](https://github.com/nimblehq/tpc-delivery-android/blob/1c53bb393d0ce842bd6b46e292d723b20793c0fc/codemagic.yaml#L223) main:
  - Command: `echo "RELEASE_NOTE_CONTENT="$(git log --merges --pretty=%B $(git describe --abbrev=0 --tags)..HEAD | grep "\[")""`
  - Sample release notes:
    ```
    [sc-63766] [Integrate] As a user, I can pull to refresh to get the most recent data on the Payment screen [sc-63767] [UI] As a user, I can pull to refresh to get the most recent data on the Payment screen [sc-59154] [sc-60298] [UI + Integrate] As a user, I can see a message if my history is empty [sc-63706] [Integrate] As a user, I see a toast message to understand that the download is in progress [sc-63422] Pin Improve the padding between the horizontal avatar list & the timer info in the Document Detail screen [sc-62943] [UI] As a user, I can see the recent activities on the home screen [sc-63424] Improve the bottom nav bar on the Main screen to show the tab text properly [Chore] Bump app version 0.8.0
    ```

## Who Benefits?

iOS folks

## What's Next?

- [x] Vote to go with the new feature proposal or not
- [x] Create the issue to iOS Template and start working on it if the majority vote for **YES**

**Comment by markgravity** [comment]
Nice one, I already did something similar in some client projects, it comes from the PM want to include the task ID in the release note so they can know what to test 🤣 

But I think your way is much better and simple
```
def get_release_notes()
  	release_notes = ""
    commit = @fastlane.last_git_commit
    message = commit[:message]
    lines = message.split(/(\n)/)
    if lines[4]
      release_notes = lines[4]
    end
    return release_notes
  end
```

**Comment by minhnimble** [comment]
Since we had almost half of the votes as __YES__ and the voting has opened for 1 month, we will process the next step with a new [issue](https://github.com/nimblehq/ios-templates/issues/493) and wrap up this discussion, thank you for your voting 🙏
